addons/mail/models/mail_thread.py

```diff
@@ -2135,6 +2135,41 @@ def _message_create(self, values_list):
     # NOTIFICATION API
     # ------------------------------------------------------
 
+    @api.model
+    def discard_notifications(self, notification_type):
+        if notification_type in {'email', 'sms'}:
```

Comment:
As mail and sms share the same code, I would remove the "if" and simply do by default what is herebelow. Snailmail code should go into its own module, using classic inheritance.

Reply:
Yes that's better, indeed. Done